File: php/Models/Dpr.php
<!-- Saya Muhammad Muhammad Fadlul Hafiizh [2209889] mengerjakan soal latprak_1 dalam mata kuliah DPBO.
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan, Aamiin -->
<?php
class Dpr{
    private $id;
    private $nama;
    private $bidang;
    private $partai;
    private $foto_url;

    //constructor
    public function __construct($nama = "", $bidang="", $partai="") {
        $this->id = uniqid();
        $this->nama = $nama;
        $this->bidang = $bidang;
        $this->partai = $partai;
    }

    //enkapsulasi semua atribut
    public function getId() {return $this->id;}
    
    public function getFoto() {return $this->foto_url;}

	public function getNama() {return $this->nama;}

	public function getBidang() {return $this->bidang;}

	public function getPartai() {return $this->partai;}

	public function setNama( $nama): void {$this->nama = $nama;}
	
    public function setFoto( $foto): void {$this->foto_url = $foto;}

	public function setBidang( $bidang): void {$this->bidang = $bidang;}

	public function setPartai( $partai): void {$this->partai = $partai;}

	public function __destruct(){}
	

}

File: java/models/Dpr.java
/*Saya Muhammad Muhammad Fadlul Hafiizh [2209889] mengerjakan soal latprak_ dalam mata kuliah DPBO.
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan, Aamiin */

package models;

import java.util.ArrayList;
import java.util.List;

/**
 * Dpr
 */
public class Dpr {
    private int id;
    private String nama;
    private String bidang;
    private String partai;
    private List<Dpr> daftarAnggota;

    //constructor
    public Dpr(int id, String nama, String bidang, String partai) {
        this.id = id;
        this.nama = nama;
        this.bidang = bidang;
        this.partai = partai;
        this.daftarAnggota = new ArrayList<>();
    }
    public Dpr(){
        this.daftarAnggota = new ArrayList<>();
    }

    //enkapsulasi semua atribut
    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public String getNama() {
        return nama;
    }
    public void setNama(String nama) {
        this.nama = nama;
    }
    public String getBidang() {
        return bidang;
    }
    public void setBidang(String bidang) {
        this.bidang = bidang;
    }
    public String getPartai() {
        return partai;
    }
    public void setPartai(String partai) {
        this.partai = partai;
    }
    public List<Dpr> getMyList() {
        return daftarAnggota;
    }

    //untuk menambahkan obj pada list
    public void addList(Dpr objDpr){
        // List<Dpr> myList = new ArrayList<>();
        this.daftarAnggota.add(objDpr);
    }

    

    
}